### PR TITLE
Workaround / fix for issues #99 and #107

### PR DESCRIPTION
**Pull request by srguiwiz**
Workaround / fix for issues #99 and #107 .

To replace pull request #100, which erroneously didn't come from a topical branch and hence has been polluted.

### CODE REVIEW

**Comment by slnode** [comment]
Can one of the admins verify this patch? To accept patch and trigger a build add comment ".*ok\W+to\W+test.*"

**Comment by bajtos** [comment]
@slnode ok to test

**Comment by bajtos** [comment]
@srguiwiz thank you for the pull request. I am not enough familiar with this code base, your pull request has to wait for @raymondfeng to review it.

**Comment by ebourmalo** [comment]
any news update this @bajtos ??

**Comment by slnode** [comment]
Can one of the admins verify this patch? To accept patch and trigger a build add comment ".*ok\W+to\W+test.*"

File: lib/mongodb.js [hunk 1]
```diff
@@ -486,7 +486,7 @@ MongoDB.prototype.buildWhere = function (model, where) {
       }
       else {
         query[k] = {};
-        query[k]['$' + spec] = cond;
+        query[k][spec[0] !== '$' ? '$' + spec : spec] = cond;
```
Comment (raymondfeng) [inline lib/mongodb.js]: I wonder if you should check `allowExtendedOperators ` setting as https://github.com/srguiwiz/loopback-connector-mongodb/blob/workaround-issues-99-and-107/lib/mongodb.js#L321.

**Comment by bajtos** [comment]
@ebourmalo I am afraid this patch looks rather abandoned. Would you mind to take it over, address Raymond's comment above and submit a new pull request? Also please include unit-tests to verify the change works as intended and prevent regressions in the future.

**Comment by ssh24** [comment]
Issue #99 and #107 are not bugs as of the current released connector (`v3.1.0`). Closing this PR.

**Pull request closed without merging**